### ISSUE TITLE
Does not track dynamically created targets

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/sentinel-dom#8]
### I would like to submit
- [x] a bug
> Note: Please use a [Pull request](https://github.com/kettanaito/sentinel-dom/pulls) for submitting a new feature request.

### Environment
**Sentinel DOM version:** 0.7.8
**OS:** MacOS Sierra
**Node version:** 7
**NPM version:** 5

### Current behavior
- DOM elements created after the Tracker definition are not tracked, even if they match the provided `targets` criteria

### Expected behavior
- DOM elements created after the Tracker definition, but matching the `targets` criteria should be tracked

### Steps to reproduce
1. Declare a new Tracker.
2. Create a new DOM element matching the `targets` criteria of the Tracker
3. Scroll so the target should resolve.

**Comment by kettanaito** [comment]
## Update
This issue is caused by the current way of storing the tracking targets. The targets get converted into an Array on the `constructor`. This encapsulates them, breaking a dynamic reference which is a `targets` option.

## TL;DR
Only the `targets` existing by the time `constructor` is called are tracked.

## Solution
Ensure that `targets` are iterable within the `track` method. This way the current targets are converted into an Array, including the dynamically created targets.

## Downsides
- May affect the performance, since now it's needed to ensure targets are iterable within each tracking attempt

**Comment by kettanaito** [comment]
Solution is already provided, needs testing.

**Comment by kettanaito** [comment]
## Update
Test is written and passing. !20.
Closing.
